src/system.rs

```diff
@@ -164,21 +210,28 @@ pub type TypeId = usize;
 /// Defines a cache that can be used to store resolved [Type]s of a [Schema]
 #[derive(Debug, Clone)]
 pub struct TypeStore {
+    imports: HashMap<String, TypeId>, // stores all the imported types of a schema
     ids_by_name: HashMap<String, TypeId>,
     types_by_id: Vec<TypeDefinition>,
 }
 
 impl TypeStore {
     pub fn new() -> Self {
         Self {
+            imports: HashMap::new(),
             ids_by_name: HashMap::new(),
             types_by_id: Vec::new(),
         }
     }
 
-    /// Returns [Type]s stored in the [TypeStore] to be used by [SchemaTypeIterator]
-    pub fn get_types(&self) -> &[TypeDefinition] {
-        &self.types_by_id
+    /// Returns [TypeId]s stored in the [TypeStore] to be used by [SchemaTypeIterator]
+    pub fn get_types(&self) -> Vec<TypeId> {
+        self.ids_by_name.values().cloned().collect()
```

Comment:
Ahh, now I see why you added a `Vec` to the iterator. You could skip that by having this method return an iterator over `self.ids_by_name.values()` and storing that in the `SchemaTypeIterator`. (This would involve adding a lifetime, but I think the blast radius is small?)

Reply:
Reference issue: https://github.com/amzn/ion-schema-rust/issues/40